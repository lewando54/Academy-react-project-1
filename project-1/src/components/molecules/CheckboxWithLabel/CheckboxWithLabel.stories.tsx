/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import CheckboxWithLabel from './CheckboxWithLabel'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: CheckboxWithLabel,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    color: { control: 'color' },
    id: { control: 'text' },
    children: { control: 'text' }
  }
} as ComponentMeta<typeof CheckboxWithLabel>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof CheckboxWithLabel> = (args) => <CheckboxWithLabel {...args} />

export const Primary = Template.bind({})

Primary.args = {
  color: '#e90800',
  children: 'Example label',
  id: 'checkbox-id'
}
// More on args: https://storybook.js.org/docs/react/writing-stories/args
