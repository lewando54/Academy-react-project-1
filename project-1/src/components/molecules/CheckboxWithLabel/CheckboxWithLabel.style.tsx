import styled from 'styled-components'

export default styled.div`
    width: 100%;
    display: flex;
    gap: 0.5rem;
    flex-direction: row-reverse;
    justify-content: flex-end;
    align-items: center;
`
