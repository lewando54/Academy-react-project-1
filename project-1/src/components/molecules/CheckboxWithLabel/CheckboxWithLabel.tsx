import React, { forwardRef } from 'react'
import Checkbox, { CheckboxProps } from '../../atoms/Checkbox/Checkbox'
import Label, { LabelProps } from '../../atoms/Label/Label'
import CheckboxWithLabelWrapper from './CheckboxWithLabel.style'

type CheckboxWithLabelProps = CheckboxProps & LabelProps

function CheckboxWithLabel (props: CheckboxWithLabelProps, ref: React.Ref<HTMLInputElement>): React.ReactElement {
  return (
        <CheckboxWithLabelWrapper>
            <Label htmlFor={props.id} testId={props.testId}>{props.children}</Label>
            <Checkbox color={props.color} ref={ref} id={props.id} testId={props.testId} />
        </CheckboxWithLabelWrapper>
  )
}

export default forwardRef(CheckboxWithLabel)
