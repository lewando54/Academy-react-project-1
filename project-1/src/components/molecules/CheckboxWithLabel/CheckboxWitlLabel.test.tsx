import React from 'react'
import '@testing-library/jest-dom'
import * as renderer from 'react-test-renderer'
import CheckboxWithLabel from './CheckboxWithLabel'

describe('CheckboxWithLabel', () => {
    // SprawdziÄ‡ children w przekazaniu
  it('should render correctly', () => {
    const tree = renderer.create(<CheckboxWithLabel htmlFor='checkid' id='checkid' testId='test'>Testing</CheckboxWithLabel>)
    expect(tree).toMatchSnapshot()
  })
})
