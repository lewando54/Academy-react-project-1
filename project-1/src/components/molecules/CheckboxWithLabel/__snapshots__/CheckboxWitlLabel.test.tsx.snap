// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel should render correctly 1`] = `
<div
  className="sc-gswNZR iUYsYD"
>
  <label
    data-testid="test"
    htmlFor="checkid"
  >
    Testing
  </label>
  <input
    className="sc-bcXHqe gGsKtD"
    data-testid="test"
    id="checkid"
    type="checkbox"
  />
</div>
`;
